dashboard/app/controllers/featured_projects_controller.rb

```diff
@@ -1,23 +1,18 @@
 class FeaturedProjectsController < ApplicationController
-  def create
-    unless current_user.try(:permission?, UserPermission::PROJECT_VALIDATOR)
-      _, channel_id = storage_decrypt_channel_id(featured_project_params[:project_id])
-      @featured_project = FeaturedProject.create({storage_app_id: channel_id, who_featured_user_id: current_user.id})
-      @featured_project.save!
-    end
-  end
+  authorize_resource
 
-  def destroy_by_project_id
-    unless current_user.try(:permission?, UserPermission::PROJECT_VALIDATOR)
-      _, channel_id = storage_decrypt_channel_id(params[:project_id])
-      @featured_project = FeaturedProject.find_by_storage_app_id(channel_id)
-      @featured_project.destroy
-    end
+  def feature
+    _, channel_id = storage_decrypt_channel_id(params[:project_id])
+    return render_404 unless channel_id
+    @featured_project = FeaturedProject.find_or_create_by!(storage_app_id: channel_id)
+    @featured_project.update! unfeatured_at: nil
+    @featured_project.update! featured_at: DateTime.now
```

Comment:
nit: these should be combined into a single `update!` as suggested [here](https://github.com/code-dot-org/code-dot-org/pull/20348#discussion_r165785133), or 2 local assignments and a single save. Each `update!` commits to the DB, so this will generate an extra db transaction. Here it's not a big deal since this won't be a high scale API, but still good practice.